controllers/sriovoperatorconfig_controller.go

```diff
@@ -94,58 +94,89 @@ func (r *SriovOperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	snolog.SetLogLevel(defaultConfig.Spec.LogLevel)
 
-	r.FeatureGate.Init(defaultConfig.Spec.FeatureGates)
-	logger.Info("enabled featureGates", "featureGates", r.FeatureGate.String())
+	// examine DeletionTimestamp to determine if object is under deletion
+	if defaultConfig.ObjectMeta.DeletionTimestamp.IsZero() {
 
-	if !defaultConfig.Spec.EnableInjector {
-		logger.Info("SR-IOV Network Resource Injector is disabled.")
-	}
+		if !sriovnetworkv1.StringInArray(sriovnetworkv1.OPERATORCONFIGFINALIZERNAME, defaultConfig.ObjectMeta.Finalizers) {
```

Comment:
this seems overly complex to what we are trying to achieve IMO. plus we are now trying to `r.deleteAllWebhooks()`
twice : L87 and L166

IMO we should:

1. if object not found - do nothing
2. if object found - add finalizer
3. if deletion timestamp is set : call `r.deleteAllWebhooks()` and remove finalizer.
4. do the regular flow as it is today.

Reply:
Ok. I'll remove initial `r.deleteAllWebhooks()` and keep one that is being called during object deletion